docs/reference/search/request/highlighting.asciidoc

```diff
@@ -7,6 +7,11 @@ When you request highlights, the response contains an additional `highlight`
 element for each search hit that includes the highlighted fields and the
 highlighted fragments.
 
+NOTE: For some complex queries (e.g nested boolean queries, queries using
+`minimum_should_match` etc.), parts of documents may be highlighted
+that don't correspond to query matches. The work for fixing this is
+currenly in progress.
+
```

Comment:
I think you should remove this part or say something like `highlighters don't reflect the boolean logic of the query when extracting the terms to highlight...`. I am not sure that we're going to "fix" it and we should not add this statement in the docs. I see pros and cons to do that and the solution that @romseygeek implemented might not be applicable in all cases (term vectors highlighting for instance).

Reply:
@jimczi Thanks, Jim. I will rephrase this note as you suggested, and remove the part about fixing.  I was asked to create this note, as there were several SDH and other issues, where highlighted fragments did not match a query, which confused users.